pkg/k8s/client.go

```diff
@@ -105,11 +109,66 @@ func CreateConfig() (*rest.Config, error) {
 	return createConfig(GetAPIServerURL(), GetKubeconfigPath(), GetQPS(), GetBurst())
 }
 
+func setDialer(config *rest.Config) func() {
+	context := (&net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}).DialContext
+	dialer := connrotation.NewDialer(context)
+	config.Dial = dialer.DialContext
+	return dialer.CloseAll
+}
+
+func createHeartbeatClient() (*rest.RESTClient, func(), error) {
+	restConfig, err := CreateConfig()
+	if err != nil {
+		return nil, nil, err
+	}
+	timeout := option.Config.K8sHeartbeatTimeout
+	restConfig.ContentConfig.NegotiatedSerializer = runtime.NewSimpleNegotiatedSerializer(runtime.SerializerInfo{})
+	restConfig.Timeout = timeout
+	closeAllConns := setDialer(restConfig)
+	restClient, err := rest.UnversionedRESTClientFor(restConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+	return restClient, closeAllConns, nil
+}
+
+func runHeartbeat(closeAllConns []func(), restClient *rest.RESTClient, stop chan struct{}) error {
+	timeout := option.Config.K8sHeartbeatTimeout
+	go wait.Until(func() {
+		done := make(chan error)
+		go func() {
+			err := restClient.Get().AbsPath("/healthz").Do().Error()
+			if err != nil {
+				done <- err
+			}
+
+			close(done)
+		}()
```

Comment:
I'm afraid of this. It seems harmless but that won't be the case for big clusters. `/healthz` could return a 503 and we would restart the connection, possibly make it worse.

Reply:
How does kubelet cope with this?

Reply:
Kubelet does a get on nodes